worlds/stardew_valley/test/rules/TestSkills.py

```diff
@@ -25,16 +25,56 @@ def test_all_skill_levels_require_previous_level(self):
         for skill in all_skills:
             self.collect_everything()
             self.remove_by_name(f"{skill} Level")
+
             for level in range(1, 11):
                 location_name = f"Level {level} {skill}"
+                location = self.multiworld.get_location(location_name, self.player)
+
                 with self.subTest(location_name):
-                    can_reach = self.can_reach_location(location_name)
                     if level > 1:
-                        self.assertFalse(can_reach)
+                        self.assert_reach_location_false(location, self.multiworld.state)
                         self.collect(f"{skill} Level")
-                        can_reach = self.can_reach_location(location_name)
-                    self.assertTrue(can_reach)
-            self.multiworld.state = self.original_state.copy()
 
+                    self.assert_reach_location_true(location, self.multiworld.state)
+
+            self.reset_collection()
+
+
+class TestMasteryRequireSkillBeingMaxed(SVTestBase):
```

Comment:
Can we make some tests for SkillProgression.vanilla, and maybe even with progressive?
The masteries rule are somewhat strict on all settings, so I think it would have value to check these as well. We've had lots of bugs in the past with settings turned off having wrong logic expectations